From f4f3123974b03c4c7d671ce36f41bef336b1a8a6 Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Sun, 6 Mar 2016 22:08:15 +0100
Subject: Revert "wined3d: Call wined3d_texture_set_dirty() in
 wined3d_volume_invalidate_location()."

This reverts commit b23f21a9fb6e09afad6ac3327233130f8be979b2.
---
 dlls/wined3d/volume.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/dlls/wined3d/volume.c b/dlls/wined3d/volume.c
index 457067d..2d88af6 100644
--- a/dlls/wined3d/volume.c
+++ b/dlls/wined3d/volume.c
@@ -109,11 +109,7 @@ void wined3d_volume_validate_location(struct wined3d_volume *volume, DWORD locat
 void wined3d_volume_invalidate_location(struct wined3d_volume *volume, DWORD location)
 {
     TRACE("Volume %p, clearing %s.\n", volume, wined3d_debug_location(location));
-
-    if (location & (WINED3D_LOCATION_TEXTURE_RGB | WINED3D_LOCATION_TEXTURE_SRGB))
-        wined3d_texture_set_dirty(volume->container);
     volume->locations &= ~location;
-
     TRACE("new location flags are %s.\n", wined3d_debug_location(volume->locations));
 }
 
@@ -556,7 +552,10 @@ HRESULT wined3d_volume_map(struct wined3d_volume *volume,
     }
 
     if (!(flags & (WINED3D_MAP_NO_DIRTY_UPDATE | WINED3D_MAP_READONLY)))
+    {
+        wined3d_texture_set_dirty(texture);
         wined3d_volume_invalidate_location(volume, ~volume->resource.map_binding);
+    }
 
     volume->resource.map_count++;
 
-- 
2.7.1

